features/keychain/module/keychain.js

```diff
@@ -42,6 +44,34 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
+  hasLockedPrivateKeys() {
+    return this.#lockedPrivateKeys
+  }
+
+  lockPrivateKeys() {
+    this.#lockedPrivateKeys = true
+  }
+
+  unlockPrivateKeys(seeds) {
+    assert(this.#lockedPrivateKeys, 'already unlocked')
+    assert(
+      seeds?.length === Object.values(this.#masters).length,
+      'must pass in same number of seeds'
+    )
+    for (const seed of seeds) {
+      const seedId = getSeedId(seed)
+      assert(!!this.#masters[seedId], 'must pass in existing seed')
+    }
+
+    this.removeAllSeeds()
+    this.#lockedPrivateKeys = false
+    return this.addSeeds(seeds)
```

Comment:
why do we need to remove and r-eadd if we validate they're the same?

Reply:
I guess we don't have to & could just unlock instead

Reply:
I was preparing the code for the case where we would only keep public keys. But since that is not an option anymore we can drop it

Reply:
334af833f39d0c29e19f61c0ec71ee42724c34db